drivers/regulator/regulator_axp2101.c

```diff
@@ -0,0 +1,557 @@
+/*
+ * Copyright (c) 2021 NXP
+ * Copyright (c) 2023 Martin Kiepfer <mrmarteng@teleschirm.org>
+ * Copyright (c) 2024 Deng Baoan <dengbaoan@foxmail.com>
+ * SPDX-License-Identifier: Apache-2.0
+ */
+
+#define DT_DRV_COMPAT x_powers_axp2101_regulator
+
+#include <errno.h>
+#include <zephyr/kernel.h>
+#include <zephyr/drivers/i2c.h>
+#include <zephyr/drivers/regulator.h>
+#include <zephyr/sys/linear_range.h>
+#include <zephyr/sys/util.h>
+#include <zephyr/logging/log.h>
+#include <zephyr/logging/log_instance.h>
+
+LOG_MODULE_REGISTER(regulator_axp2101, CONFIG_REGULATOR_LOG_LEVEL);
+
+#define XPOWERS_AXP2101_DC_ONOFF_DVM_CTRL                (0x80)
```

Comment:
Please create a header file in include/zephyr/drivers/mfd/
and move these definitions in the same as done in AXP192.